actors/builtin/miner/deadline_state.go

```diff
@@ -1065,3 +1055,56 @@ func (dl *Deadline) RecordProvenSectors(
 		RetractedRecoveryPower: retractedRecoveryPowerTotal,
 	}, nil
 }
+
+func (dl *Deadline) RescheduleSectorExpirations(
+	store adt.Store, sectors Sectors, ssize abi.SectorSize, quant QuantSpec,
+	expiration abi.ChainEpoch, partitionSectors PartitionSectorMap,
+) error {
+	partitions, err := dl.PartitionsArray(store)
+	if err != nil {
+		return err
+	}
+
+	var rescheduledPartitions []uint64 // track partitions with moved expirations.
+	if err := partitionSectors.ForEach(func(partIdx uint64, sectorNos *abi.BitField) error {
+		var partition Partition
+		if found, err := partitions.Get(partIdx, &partition); err != nil {
+			return xc.ErrIllegalState.Wrapf("failed to load partition %d: %w", partIdx, err)
+		} else if !found {
+			// We failed to find the partition. That's ok, we'll skip it.
```

Comment:
Why would this be ok? Could you give more context as to a state that could cause this? Is it due to compaction and the best-effort deadline-partition index queue?

Reply:
See the new comments.